angular/projects/admin-nrpti/src/app/mines/mines-records-list/mines-records-list.component.ts

```diff
@@ -413,6 +466,37 @@ export class MinesRecordsListComponent implements OnInit, OnDestroy {
     );
   }
 
+  /**
+   * Cancel adding records to collection.
+   *
+   * @memberof MinesRecordsListComponent
+   */
+  cancelAddEditCollectionRecords() {
+    const shouldCancel = confirm(
+      'Leaving this page will discard unsaved changes. Are you sure you would like to continue?'
+    );
+    if (shouldCancel) {
+      if (this.storeService.getItem(StateIDs.collectionAddEdit).collectionId) {
+        this.router.navigate(['mines', this.mine._id, 'collections', this.collectionAddEditState.collectionId, 'edit']);
+      } else {
+        this.router.navigate(['mines', this.mine._id, 'collections', 'add']);
+      }
+    }
+  }
+
+  /**
+   * Submit adding records to collection.
+   *
+   * @memberof MinesRecordsListComponent
+   */
+  submitAddEditCollectionRecords() {
```

Comment:
Method is fine, but I think we'll need a discussion about the behavior with this button click. You're just following the spec so no criticism here. But. When you come into the form, you have no way of knowing if the click will route you to the `add collection` state, or the `edit collection` state. You have no identifier that a collection has already been 'selected' if you came in from the collection screen, and no way to clear a collection if you wanted to.

Because the breadcrumbs are not helpful, it only makes it worse as you have no idea, at a glance, how you got to this screen if you're coming from a collection. It's very un-intuitive and definitely needs more UI cues as to what is going on. I was confused and I knew what I was doing.